### ISSUE TITLE
SPV causing nodes to randomly segfault

### ISSUE THREAD

**Issue by prasannavl** [DeFiCh/ain#1129]
Nodes running with spv enabled (default), has been observed to be crashing over the last few weeks at random times. Restarting the node with spv=0 makes the node continue to work as normal. 

This appears due to corruption in the data, that the spv code can't seem to handle. But requires further investigation on why this is being caused in the first place, and hasn't surfaced until now -- since there has been no recent changes to this part of the related areas. 

**Stack trace:** 

```
[Current thread is 1 (Thread 0x7f6a3869b700 (LWP 607982))]
(gdb) bt
#0  0x0000556b8ca8202d in BRPeerSendMessage(BRPeerStruct*, unsigned char const*, unsigned long, char const*) ()
#1  0x0000556b8ca8466a in BRPeerSendGetdata(BRPeerStruct*, UInt256 const*, unsigned long, UInt256 const*, unsigned long) ()
#2  0x0000556b8ca99d04 in _BRPeerManagerRequestUnrelayedTx(BRPeerManagerStruct*, BRPeerStruct*) ()
#3  0x0000556b8ca999bc in _mempoolDone(void*, int) ()
#4  0x0000556b8ca88ee8 in _BRPeerAcceptMessage(BRPeerStruct*, unsigned char const*, unsigned long, char const*) ()
#5  0x0000556b8ca81154 in _peerThreadRoutine(void*) ()
#6  0x0000556b8ca8d65e in boost::detail::thread_data<boost::_bi::bind_t<void*, void* (*)(void*), boost::_bi::list1<boost::_bi::value<BRPeerStruct*> > > >::run() ()
#7  0x0000556b8cb868ba in thread_proxy ()
#8  0x00007f6a954a8609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#9  0x00007f6a95077163 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Workaround**: Disable spv with `-spv=0` flag or `spv = 0` in defi.conf. 
**Current workaround with spv turned on**: None.

**Comment by prasannavl** [comment]
**Status update**: The cause has been identified to be a bug in the third party dependency - bread wallet SPV - that uses variable length arrays in the stack. PR #1127 already fixes this for v3.  

Back-port in progress.

**Comment by prasannavl** [comment]
Resolved by #1131

**Comment by prasannavl** [comment]
v2.6.2 Released.
